aten/src/ATen/native/cpu/ReduceOpsKernel.cpp

```diff
@@ -115,9 +115,9 @@ static void cumprod_cpu_kernel(const Tensor& result, const Tensor& self, int64_t
 // custom min and max to be used in logcumsumexp for complex arguments
 template <typename scalar_t, bool min>
 c10::complex<scalar_t> _logcumsumexp_minmax(c10::complex<scalar_t> x, c10::complex<scalar_t> y) {
-  if (std::isnan(y)) {  // either real is nan or imag is nan
```

Comment:
I think this must have started failing due to removal of `isnan` from complex utils.

I was wondering why do we have `at::_isnan`? Does it do anything different from `std::isnan`?

Reply:
> I think this must have started failing due to removal of isnan from complex utils.

Nahh, these failures were earlier too. I'll work on these.

> I was wondering why do we have `at::_isnan`? Does it do anything different from `std::isnan`?

Looks like `at::_isnan` was added because std::isnan wasn't performant for integral type. Ref: https://github.com/pytorch/pytorch/blob/95dd88a73ddb40fcaaf8614e8bf87d6e185202c3/aten/src/ATen/NumericUtils.h#L17#L19. 
PR: #18021